library/havege.c

```diff
@@ -158,10 +159,11 @@
  */
 static void havege_fill( mbedtls_havege_state *hs )
 {
-    int i, n = 0;
-    int  U1,  U2, *A, *B, *C, *D;
-    int PT1, PT2, *WALK, RES[16];
-    int PTX, PTY, CLK, PTEST, IN;
+    size_t n = 0;
+    unsigned i;
```

Comment:
Why should `i` not be `size_t`? It's used to index `RES`, as in `ONE_ITERATION` (`RES[(i - 8) ^ PTY]`), and array indices tend to be `size_t`.

Reply:
`i` only takes values from 0 to 15 and is used in an inner loop, so it's more correct to have it of “the most efficient type” than “the correct type for indexing arrays”. But it isn't worth giving the reader pause for that. The compiler can easily recognize what the actual range of `i` is and optimize accordingly anyway.

Reply:
Why not `uint_fast8_t`?

Reply:
We don't officially require C99 (well, I think we made a blog post that says we do, but we still test with non-C99 compilers). We use `stdint.h` but we only assume that it provides the “basic” types: `uintNN_t`. Are we ok with adding a requirement for `uint_ADJECTIVENN_t`? Should we start using it in more places?